spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -976,6 +974,115 @@ public Message<O> process(Message<I> msg) {
 The implementation is reponsible for enforcing the acknowledgement strategy defined by the user when the `@Acknowledgement` policy is used. 
 If the annotation is not used, the default policy must be enforced.
 
+=== Connector
+
+Reactive Messaging connects matching `@Incoming` and `@Outgoing` running inside the same application.
+Besides, it maps specific _channels_ to _external_ technologies such as Apache Kafka, MQTT, Web Sockets, or JMS. 
+It means that Reactive Messaging can receive messages from virtually any messaging technology and dispatch messages to any messaging technology.
+The component responsible for this role is called _connector_. 
+
+==== Connector concepts
+
+Each _connector_ is responsible for a specific technology. 
+A connector can:
+
+* act as a _Publisher_, meaning it retrieves or receives messages from an external messaging technology. The messages will then be sent to a method annotated with `@Incoming`
+* act as a _Subscriber_, meaning it dispatches messages to an external messaging technology. The messages are received from a method annotated with `@Outgoing`
+* handle both directions
+
+It's essential that connectors implement the back-pressure protocol defined by the Reactive Streams specification.
+
+A _connector_ is implemented as a CDI Bean, generally _application_ scoped implementing:
+
+* the `org.eclipse.microprofile.reactive.messaging.connector.IncomingConnectorFactory` interface to receive messages from an external source;
+* the `org.eclipse.microprofile.reactive.messaging.connector.OutgoingConnectorFactory` interface to dispatch messages to an external sink
+
+The bean is a factory called by the Reactive Messaging implementation to create `PublisherBuilder` or `SubscriberBuilder` objects. 
+These objects are then connected to methods annotated with `@Incoming` or `@Outgoing`. 
+
+Both interfaces contains a `type()` method returning a `MessagingProvider`.
+`MessagingProvider` is a marker interface implemented by the connector allowing the user to indicate which connector needs to be used. 
+For instance, the following `MessagingProvider` implementation can be used in a Kafka connector:
+
+[source, java]
+----
+package org.eclipse.reactive.sample.kafka;
+
+import org.eclipse.microprofile.reactive.messaging.MessagingProvider;
+
+public class Kafka implements MessagingProvider {
+
+  // Just a marker class.
+
+}
+----
+
+Once defined, the user can, in the configuration, refer to this connector using the fully qualified name of the class.
```

Comment:
I would like to discuss at the hangout if it is advantageous for this to have
get...ConnectionFactory methods to reduce the amount of class loading
of ConnectionFactories that might not be used.